uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js

```diff
@@ -0,0 +1,485 @@
+      }, {});
+    }
+
+    /**
+     * Index leave requests by contact_id as first level
+     * and date as second level
+     *
+     * @param  {Array} leaveRequestsList
+     * @return {Promise}
+     */
+    function indexLeaveRequests (leaveRequestsList) {
+      var deferred = $q.defer();
+
+      leaveRequestsList.forEach(function (leaveRequest) {
+        var days = leaveRequestDays(leaveRequest);
+
+        leaveRequests[leaveRequest.contact_id] = leaveRequests[leaveRequest.contact_id] || {};
+
+        days.forEach(function (day) {
+          leaveRequests[leaveRequest.contact_id][day.date] = leaveRequest;
+        });
+      });
+
+      deferred.resolve();
+
+      return deferred.promise;
+    }
+
+    /**
+     * Initializes the event listeners
+     */
+    function initListeners () {
+      $rootScope.$on('LeaveCalendar::showMonths', showMonthIfInList);
+      $rootScope.$on('LeaveRequest::new', addLeaveRequest);
+      $rootScope.$on('LeaveRequest::edit', updateLeaveRequest);
+      $rootScope.$on('LeaveRequest::updatedByManager', updateLeaveRequest);
+      $rootScope.$on('LeaveRequest::deleted', deleteLeaveRequest);
+    }
+
+    /**
+     * Returns whether a date is of a specific type
+     * half_day_am or half_day_pm
+     *
+     * @param  {string} name
+     * @param  {object} leaveRequest
+     * @param  {string} date
+     *
+     * @return {boolean}
+     */
+    function isDayType (name, leaveRequest, date) {
+      var dayType = vm.supportData.dayTypes[name];
+
+      if (moment(date).isSame(leaveRequest.from_date)) {
+        return dayType.value === leaveRequest.from_date_type;
+      }
+
+      if (moment(date).isSame(leaveRequest.to_date)) {
+        return dayType.value === leaveRequest.to_date_type;
+      }
+    }
+
+    /**
+     * Returns whether a leaveRequest is pending approval
+     *
+     * @param  {object} leaveRequest
+     * @return {boolean}
+     */
+    function isPendingApproval (leaveRequest) {
+      var status = vm.supportData.leaveRequestStatuses[leaveRequest.status_id];
+
+      return status.name === sharedSettings.statusNames.awaitingApproval;
+    }
+
+    /**
+     * Decides whether sent date is a public holiday
+     *
+     * @param  {string} date
+     * @return {boolean}
+     */
+    function isPublicHoliday (date) {
+      return !!vm.supportData.publicHolidays[dateObjectWithFormat(date).valueOf()];
+    }
+
+    /**
+     * Loads the work pattern calendar and the leave request of the month,
+     * then it process the data onto each day of the month
+     *
+     * @return {Promise}
+     */
+    function loadMonthData () {
+      vm.month.loading = true;
+
+      return $q.all([
+        loadMonthWorkPatternCalendars(),
+        loadMonthLeaveRequests()
+      ])
+      .then(function () {
+        return setMonthDaysContactData();
+      })
+      .then(function () {
+        dataLoaded = true;
```

Comment:
the first is a flag to see if the month's data is already loading, the second is a boolean used in the View (as indicated by the fact that it's on the scope)